src/libraries/System.Private.CoreLib/src/System/Numerics/BFloat16.cs

```diff
@@ -0,0 +1,2011 @@
+        }
+
+        /// <summary>
+        /// Returns a string representation of the current value using the specified <paramref name="format"/> and <paramref name="provider"/>.
+        /// </summary>
+        public string ToString([StringSyntax(StringSyntaxAttribute.NumericFormat)] string? format, IFormatProvider? provider)
+        {
+            return Number.FormatFloat(this, format, NumberFormatInfo.GetInstance(provider));
+        }
+
+        /// <summary>
+        /// Tries to format the value of the current BFloat16 instance into the provided span of characters.
+        /// </summary>
+        /// <param name="destination">When this method returns, this instance's value formatted as a span of characters.</param>
+        /// <param name="charsWritten">When this method returns, the number of characters that were written in <paramref name="destination"/>.</param>
+        /// <param name="format">A span containing the characters that represent a standard or custom format string that defines the acceptable format for <paramref name="destination"/>.</param>
+        /// <param name="provider">An optional object that supplies culture-specific formatting information for <paramref name="destination"/>.</param>
+        /// <returns></returns>
+        public bool TryFormat(Span<char> destination, out int charsWritten, [StringSyntax(StringSyntaxAttribute.NumericFormat)] ReadOnlySpan<char> format = default, IFormatProvider? provider = null)
+        {
+            return Number.TryFormatFloat(this, format, NumberFormatInfo.GetInstance(provider), destination, out charsWritten);
+        }
+
+        /// <inheritdoc cref="IUtf8SpanFormattable.TryFormat" />
+        public bool TryFormat(Span<byte> utf8Destination, out int bytesWritten, [StringSyntax(StringSyntaxAttribute.NumericFormat)] ReadOnlySpan<char> format = default, IFormatProvider? provider = null)
+        {
+            return Number.TryFormatFloat(this, format, NumberFormatInfo.GetInstance(provider), utf8Destination, out bytesWritten);
+        }
+
+        //
+        // Explicit Convert To BFloat16
+        //
+
+        /// <summary>Explicitly converts a <see cref="char" /> value to its nearest representable <see cref="BFloat16"/> value.</summary>
+        /// <param name="value">The value to convert.</param>
+        /// <returns><paramref name="value" /> converted to its nearest representable <see cref="BFloat16"/> value.</returns>
+        public static explicit operator BFloat16(char value) => (BFloat16)(float)value;
+
+        /// <summary>Explicitly converts a <see cref="decimal" /> value to its nearest representable <see cref="BFloat16"/> value.</summary>
+        /// <param name="value">The value to convert.</param>
+        /// <returns><paramref name="value" /> converted to its nearest representable <see cref="BFloat16"/> value.</returns>
+        public static explicit operator BFloat16(decimal value) => (BFloat16)(float)value;
```

Comment:
Do we have any concern of double rounding here?

For others, like `char->float->bfloat16` we know that we're getting an infinitely precise `float`. However, for `decimal->float->bfloat16` we're losing precision and then losing it again. Presumably the first loss of precision could result in some midpoint value?

Reply:
Maybe. Since `decimal`->`float` is also implemented in software, it should be feasible to port the algorithm.

Reply:
Currently `decimal`->`float` is already implemented as `decimal`->`double`->`float`. I think the scope is too large for now.

Reply:
Opened #112474